liquid-rust-typeck/src/lib.rs

```diff
@@ -96,17 +99,20 @@ impl<'tcx, 'a> Checker<'tcx, 'a> {
             tcx,
             global_env: task.global_env,
             bb_envs,
+            body: &task.body,
             ret_env,
             local_gen,
         };
 
+        let mut seen = HashSet::new();
```

Comment:
I'd add `seen` as a field on `Checker` and pass `self` mutably.

Reply:
Adding `seen` as a field on `Checker` messes with some borrow checking stuff - in particular, in `check_basic_block`, we can't borrow the `bb_env` while also doing the stuff in `enter_basic_block` (the closure needs "exclusive access to `self`").